### ISSUE TITLE
With `--enable=frozen-string-literal` in $RUBYOPT, Yardoc breaks.

### ISSUE THREAD

**Issue by jdickey** [lsegal/yard#1040]
The `frozen-string-literal` Ruby feature, as I understand it, does just what it says on the tin: requires all literal string values to be treated as frozen by the Ruby (MRI) interpreter. This is, more or less, the equivalent of adding the `# frozen-string-literal: true` directive at the beginning of each source file, which is reportedly going to be the default for Ruby 3.0.

All is well until you try installing some Gems with YARD documentation; the Gem install is unaffected, but the YARD documentation breaks. I discovered this while installing [customink/stuffed_bunny](https://github.com/customink/stuffed_bunny), against which I opened customink/stuffed_bunny#5. The "steps to reproduce" below reproduce that installation.

## Steps to reproduce

1. Ensure that you have run `yard config --gem-install-yri` as recommended during the `yard` Gem installation.
2. Set the `RUBYOPT` environment variable as such: `RUBYOPT=--enable=frozen-string-literal`.
3. Run the command `gem install stuffed_bunny`.  (Any Gem should do, not only `stuffed_bunny`).

## Actual Output

```
$ gem install stuffed_bunny
Fetching: stuffed_bunny-1.0.2.gem (100%)
Successfully installed stuffed_bunny-1.0.2
Building YARD (yri) index for stuffed_bunny-1.0.2...
ERROR:  While generating documentation for stuffed_bunny-1.0.2
... MESSAGE:   can't modify frozen String
... YARDOC args: --no-progress -c -n --db /Users/jeffdickey/src/conversagence/discovery/v1/discovery_cs_v1/./gemsets/extras/doc/stuffed_bunny-1.0.2/.yardoc --quiet lib
(continuing with the rest of the installation)
Done installing documentation for stuffed_bunny after 0 seconds
1 gem installed
$
```

## Expected Output

```
$ gem install stuffed_bunny
Fetching: stuffed_bunny-1.0.2.gem (100%)
Successfully installed stuffed_bunny-1.0.2
Building YARD (yri) index for stuffed_bunny-1.0.2...
Done installing documentation for stuffed_bunny after 0 seconds
1 gem installed
$
```

## Environment details:

* OS: Mac OS X 10.11.6 El Capitan
* Ruby version (`ruby -v`): ruby 2.3.3p222 (2016-11-21 revision 56859) [x86_64-darwin15]
* YARD version (`yard -v`): yard 0.9.5
* Relevant software dependency/versions:
  * RubyGems 2.6.8

I have read the [Contributing Guide][contrib].

[contrib]: https://github.com/lsegal/yard/blob/master/CONTRIBUTING.md

**Comment by lsegal** [comment]
This should work in yard-master, but #1012 hasn't yet made it to a release.

**Comment by jdickey** [comment]
Thanks for the tip about lsegal/yard#1012; will keep an eye on that and reevaluate once it gets merged.

**Comment by lsegal** [comment]
This is released in 0.9.6.
